concordium-consensus/src/Concordium/Queries.hs

```diff
@@ -352,17 +366,17 @@ liftSkovQueryBHIAndVersion av0 av1 bhi = do
                 liftSkovQueryLatest
                     ( do
                         -- consensus version 0
-                        bp <- case other of
-                            Best -> bestBlock
-                            LastFinal -> lastFinalizedBlock
-                        (bpHash bp,) . Just <$> av0 evc bp
+                        (bp, blockContext) <- case other of
+                            Best -> (,BCBest) <$> bestBlock
```

Comment:
... And it will return all the accounts that existed at the end of that (now former) block. Which was the best block when the query started and so the `BHIQueryResponse` will yield the hash of that now former best block. 

Or do you mean that this logic is faulty as it should do something different?

Reply:
This
```haskell
-- | Get a list of all account addresses and their associated account indices.
--  There are no guarantees of the order of the list. This is because the resulting list is potentially
--  a concatenation of two lists of account addresses.
allAccounts :: (SupportsPersistentAccount pv m) => Accounts pv -> m [(AccountAddress, AccountIndex)]
allAccounts accounts = do
    persistedAccs <- LMDBAccountMap.all
    mDiffMap <- liftIO $ readIORef (accountDiffMap accounts)
    case mDiffMap of
        Nothing -> return persistedAccs
        Just accDiffMap -> do
            flattenedDiffMapAccounts <- DiffMap.flatten accDiffMap
            return $! persistedAccs <> flattenedDiffMapAccounts
```

will always return at least all the accounts that are in LMDB. Regardless of what block you give it.

And this is ultimately called. So you're claiming to return from the best block, and this is the hash you are returning in the header, but the list of accounts is from some other block.

Reply:
I changed this.